plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -341,14 +340,7 @@ else if (isFilterPushdownUnsupported(entry.getValue())) {
     // See https://docs.pinot.apache.org/developers/advanced/null-value-support for more info.
     private boolean isFilterPushdownUnsupported(Domain domain)
     {
-        ValueSet valueSet = domain.getValues();
```

Comment:
Should we change the behavior based on `enableColumnBasedNullHandling` in a spec? 
https://docs.pinot.apache.org/developers/advanced/null-value-support#store-nulls-at-ingestion-time

Reply:
Good idea! iirc it could be for any field spec (`metricFieldSpecs`, `dateTimeFieldSpecs`).

Reply:
I reverted changes in this method to minimize the scope of this PR. We can update it in follow-up.